src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingUtils.java

```diff
@@ -75,6 +76,27 @@ public class StreamingUtils {
   public static final String STREAMING_CONSTANT_OAUTH_CLIENT_SECRET = "oauth_client_secret";
   public static final String STREAMING_CONSTANT_OAUTH_REFRESH_TOKEN = "oauth_refresh_token";
 
+  // Offset verification function to verify that the current start offset has to be the previous end
+  // offset + 1, note that there are some false positives when SMT is used.
+  public static final OffsetTokenVerificationFunction offsetTokenVerificationFunction =
+      (prevBatchEndOffset, curBatchStartOffset, curBatchEndOffset, rowCount) -> {
+        boolean isMatch = true;
+        if (prevBatchEndOffset != null) {
+          try {
+            long curStart = Long.parseLong(curBatchStartOffset);
+            long prevEnd = Long.parseLong(prevBatchEndOffset);
+
+            if (curStart != prevEnd + 1) {
+              isMatch = false;
+            }
+          } catch (NumberFormatException ignored) {
+            isMatch = false;
```

Comment:
do we anticipate this happening? I would rather throw an exception since we know this is not happening. WDYT?